apps/ops/ws.py

```diff
@@ -71,8 +75,13 @@ async def receive_json(self, content, **kwargs):
         user = self.scope['user']
         user_role_ids = await self.get_current_user_role_ids(user)
         has_admin_auditor_role = bool(admin_auditor_role_ids & user_role_ids)
-
-        if not has_admin_auditor_role and task.name in self.user_tasks and task.creator != user:
+        has_perms = await self.has_perms(user, ['audits.view_joblog'])
+        # (有管理员或审计员角色) 或者 (任务是用户自己创建的 或者 有查看任务日志权限), 其他情况没有权限
+        if not (
+                has_admin_auditor_role or (
+                task.name in self.user_tasks and
+                (task.creator == user or has_perms))
+        ):
             await self.send_json({'message': 'No permission', 'task': task_id})
             return
 
```

Comment:
I have no issue with this code snippet. However, as it is a sample of the current version, there might be small changes or updates since its creation that I'm unaware of at the moment due to my knowledge cutoff date being before September 2021.

In conclusion, there doesn't seem to be any issues found in terms of coding errors or performance bottlenecks; all operations can still efficiently handle the tasks based on their current specifications and conditions provided within the context.